src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+import { enableValidation, settings, resetValidation} from "../scripts/validation.js";
+import "./index.css";
+import spotsLogoSrc from "../images/logo.svg";
+import spotsAvatarSrc from "../images/avatar.jpg";
+import spotsPencilSrc from "../images/pencil.svg";
+import spotsAddSrc from "../images/add.svg";
+import Api from "../utils/api.js";
+import { setButtonText } from "../utils/helpers.js";
+
+const spotsLogo = document.getElementById("spots-logo");
+if (spotsLogo) spotsLogo.src = spotsLogoSrc;
+
+const spotsPencil = document.getElementById("spots-pencil");
+if (spotsPencil) spotsPencil.src = spotsPencilSrc;
+
+const spotsAdd = document.getElementById("spots-add");
+if (spotsAdd) spotsAdd.src = spotsAddSrc;
+
+
+const api = new Api({
+    baseUrl: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+      authorization: "ea7006fd-eee9-4b90-adca-27b6c65eefb9",
+      "Content-Type": "application/json"
+    }
+  });
+
+
+//Profile Elements
+const profileNameElement = document.querySelector(".profile__name");
+const profileDescriptionElement = document.querySelector(".profile__description");
+const spotsAvatar = document.getElementById("spots-avatar");
+if (spotsAvatar) spotsAvatar.src = spotsAvatarSrc;
+
+api.getAppInfo()
+    .then(([userData]) => {
+        profileNameElement.textContent = userData.name;
+        profileDescriptionElement.textContent = userData.about;
+        spotsAvatar.src = userData.avatar;
+    
+}).catch(console.error);
+
+const cardList = document.querySelector(".cards__list");
+
+  api.getAllCards()
+  .then((cards) => {
+      cards.forEach((item) => {
+          if (!document.querySelector(`[data-card-id="${item._id}"]`)) {
+              const cardEl = getCardElement(item);
+              cardList.append(cardEl);
+          }
+      });
+      console.log("Cards fetched:", cards);
+  })
+  .catch(console.error);
+
+
+const cardTemplate = document.querySelector("#card-template");
+
+const formElement = document.querySelector(".popup__form");
+
+if (formElement) {
+  formElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+  
+    if (!formElement.checkValidity()) {
+      console.log("Form is invalid!");
+      return;
+    }
+  
+    const cardId = formElement.getAttribute("data-card-id");
+    const updatedCard = {
+      name: formElement.querySelector(".popup__input_type_name").value,
+      link: formElement.querySelector(".popup__input_type_url").value
+    };
+  
+    api.updateCard(cardId, updatedCard)
+      .then((updatedCard) => {
+        console.log("Card updated successfully:", updatedCard);
+      })
+      .catch(console.error);
```

Comment:
Please, delete it because it has no effect. You've found a random form element and added some code for the submit handler. It doesn't work like that